pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -329,35 +349,94 @@ func (w *Webhook) getAllLatestLibraries() []libInfo {
 	return libsToInject
 }
 
-// extractLibInfo returns the language, the image,
-// and a boolean indicating whether the library should be injected into the pod
-func (w *Webhook) extractLibInfo(pod *corev1.Pod) ([]libInfo, bool) {
+type libInfoSource int
+
+const (
+	libInfoSourceNone libInfoSource = iota
+	libInfoSourceLibInjection
+	libInfoSourceSingleStepInstrumentation
+	libInfoSourceSingleStepLangaugeDetection
+)
+
+func (s libInfoSource) injectionType() string {
+	switch s {
+	case libInfoSourceSingleStepInstrumentation, libInfoSourceSingleStepLangaugeDetection:
+		return singleStepInstrumentationInstallType
+	case libInfoSourceLibInjection:
+		return localLibraryInstrumentationInstallType
+	default:
+		return "unknown"
+	}
+}
+
+func (s libInfoSource) autodetected() bool {
+	return s == libInfoSourceSingleStepLangaugeDetection
+}
+
+func (s libInfoSource) mutatePod(pod *corev1.Pod) error {
+	_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
+		Name:  instrumentationInstallTypeEnvVarName,
+		Value: s.injectionType(),
+	})
+	return nil
+}
+
+type extractedPodLibInfo struct {
+	// libs are the libraries we are going to attempt to inject into the given pod.
+	libs []libInfo
+	// languageDetection is set when we ran/used the language-detection annotation.
+	languageDetection *libInfoLanguageDetection
+	// source is where we got the data from, used for telemetry later.
+	source libInfoSource
+}
+
+func (e extractedPodLibInfo) withLibs(l []libInfo) extractedPodLibInfo {
+	e.libs = l
+	return e
+}
+
+// extractLibInfo metadata about what library information we should be
+// injecting into the pod and where it came from.
+func (w *Webhook) extractLibInfo(pod *corev1.Pod) extractedPodLibInfo {
+	var (
+		ssiEnabled        = w.isEnabledInNamespace(pod.Namespace)
```

Comment:
I think `ssiEnabled` as a name doesn't really reflect the intent.

If single step instrumentation was disabled, this webhook wouldn't be registered at the first place, and we won't reach this point. 

Maybe `namespaceEnabled` is a better name?

Reply:
I also don't think we will reach this function if the namespace was not enabled, which means that here `ssiEnabled` will never be false.

Or am I missing something?

Reply:
We do, `isPodEligible` is a superset of `isNamespaceEnabled`, you can get here if you've opted in or out of injection injection via the annotation `admission.datadoghq.com/enabled`, the ssi flag, or `mutate_unlabeled` configs. 

So `ssi` specifically is only _one_ of the factors where we disambiguate between "lib injection" (where someone specified annotations to get this data) and "single step" where `apm.instrumentation.enabled == true`

Reply:
>If single step instrumentation was disabled, this webhook wouldn't be registered at the first place, and we won't reach this point.

The webhook is enabled regardless of if single step is on :) because of the labels and `mutate_unlabeled` configs.

Reply:
hmmmm, this is confusing to me 🤔

The wehook name is auto instrumentation, and yet, it is enabled regardless of if single step instrumentation is on? 

If single step instrumentation is off, what would the webhook do?

Reply:
I also see that the webhook is enabled only when `admission_controller.auto_instrumentation.enabled` ([here](https://github.com/DataDog/datadog-agent/blob/4c815329255671890607312a2d864bb8185ff761/pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go#L93))

But, this config is always set to true (it is not set in any place in the helm chart, see [here](https://github.com/search?q=repo%3ADataDog%2Fhelm-charts+admission_controller_auto_instrumentation_enabled&type=code))

This means the webhook is always enabled no matter what? 🤔